sdk/appcenter/src/main/java/com/microsoft/appcenter/http/HttpException.java

```diff
@@ -42,6 +49,19 @@ public HttpException(int status, @NonNull String payload) {
         this.statusCode = status;
     }
 
+    /**
+     * Init.
+     *
+     * @param status  HTTP status code.
+     * @param payload HTTP payload.
+     * @param headers HTTP responseHeaders.
+     */
+
+    public HttpException(int status, String payload, Map<String, String> headers) {
+        this(status, payload);
```

Comment:
should do the other way around, have the constructor with 2 parameters calls this(status. payload, new HashMap()), also please @NonNull on both payload and headers.

Reply:
comment was closed without resolution, call `super(getDetailMessage(status, payload));` instead of this(...).